src/codecs/avif/mod.rs

```diff
@@ -37,7 +38,20 @@ impl<W: Write> AvifEncoder<W> {
     /// necessary. When data is suitably aligned, i.e. u16 channels to two bytes, then the
     /// conversion may be more efficient.
     pub fn write_image(mut self, data: &[u8], width: u32, height: u32, color: ColorType) -> ImageResult<()> {
-        let config = self.config(color);
+        self.write_image_with_speed_quality(data, width, height, color, 100, 1)
+    }
+
+    /// Encode image data with the indicated color type, quality and speed.
+    ///
+    /// The encoder currently requires all data to be RGBA8, it will be converted internally if
+    /// necessary. When data is suitably aligned, i.e. u16 channels to two bytes, then the
+    /// conversion may be more efficient.
+    pub fn write_image_with_speed_quality(mut self, data: &[u8], width: u32, height: u32, color: ColorType, quality: u8, speed: u8) -> ImageResult<()> {
+        // Clamp quality and speed to range
+        let quality = max(1, min(quality, 100));
+        let speed = max(1, min(speed, 10));
```

Comment:
I haven't tested this but I think it should be possible to use 0 as the quality and the speed. In `libavif` quality 0 means lossless compression, speed 0 is very slow but still usable.

Reply:
Thanks for pointing that out. I foolishly assumed that it would start at 1.
I've checked out ravif and it seems that quality is 0 - 100, but speed is 1 - 10.
```
/// 0-100 scale
pub quality: u8,
/// 0-100 scale
pub alpha_quality: u8,
/// rav1e preset 1 (slow) 10 (fast but crappy)
pub speed: u8,
```

Reply:
🤔 the rav1e docs say that they do have a speed 0 setting. I'm not sure why ravif starts from 1.

https://docs.rs/rav1e/0.4.0-alpha/rav1e/prelude/struct.SpeedSettings.html#method.from_preset

Reply:
That is odd. I've just tried setting the minimum to 0 and running an encode through and nothing broke and the filesize was significantly smaller than 1, so I guess it must just be a typo in the ravif comments?

Apologies for all the back on forth with that one.